### PR TITLE
Timeout policy page created

### LINKED ISSUE #194: Describe the timeout and retry policy
Each service in radius has its own timeout/retry policy/throttle limit. Each component should have the clear error message and user hit such limit. Typically, those numbers are constantly tuned based on the data (customer feedback and telemetry).

Here is the examples of the limit we may want to document:
* Deployment engine - write / delete timeout and retry
* RP - Timeouts / retry policy for async operation per resource type

### PR DESCRIPTION
**Pull request by jasonviviano**
## Description
Created a policy page to reflect the information collected in #194 

Closes #194

### CODE REVIEW

**Comment by jasonviviano** [comment]
@jkotalik @youngbupark Let me know if the current page format makes sense from a developer perspective or if I'm missing any information whenever you have time.

File: docs/content/reference/resource-schema/resource-policies/_index.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+---
```
Comment (AaronCrawfis) [inline docs/content/reference/resource-schema/resource-policies/_index.md]: I don't think `reference/resource-schema` is the right place, as this describes API behavior, not resource schema.

Instead, I'd create a new path under `reference/api` and place this as a child under the new API section. For now we can keep the API page pretty bare bones (basically blank), and then add on additional content later as we document more of the behavior of our APIs

File: docs/content/reference/resource-schema/resource-policies/_index.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+type: docs
+title: "Resource timeout/retry policies"
```
Comment (AaronCrawfis) [inline docs/content/reference/resource-schema/resource-policies/_index.md]:
```suggestion
title: "API timeout/retry policies"
```

File: docs/content/reference/resource-schema/resource-policies/_index.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+linkTitle: "Timeout/retry policies"
+description: "Learn the timeout/retry policies of Radius resources"
```
Comment (AaronCrawfis) [inline docs/content/reference/resource-schema/resource-policies/_index.md]:
```suggestion
description: "Learn the timeout/retry policies of the Radius API"
```

File: docs/content/reference/resource-schema/resource-policies/_index.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+---
+
+## Radius services
+
+There are two main Radius services that will provide the resources described below:
+
+- [Deployment Engine]({{< ref "architecture.md#deployment-engine" >}})
+- [Resource Provider]({{< ref "architecture.md#radius-resource-provider" >}})
+
+### Deployment engine resources
```
Comment (AaronCrawfis) [inline docs/content/reference/resource-schema/resource-policies/_index.md]:
```suggestion
### Deployment engine behavior
```

File: docs/content/reference/resource-schema/resource-policies/_index.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+| Operation | Resource type | Server Timeout (Seconds) | Async Operation retry condition|
+|------|:--------:|-------------|---------|------------|
+| Default retry timeout for resource deployment | The name of your resource. | 12.5 minutes per resource | Retries on 5xx errors, 50 retry requests
+| Default request timeout | | Dependent on [UCP]({{< ref "architecture.md##universal-control-plane" >}}) |
+
+### Resource provider resources
```
Comment (AaronCrawfis) [inline docs/content/reference/resource-schema/resource-policies/_index.md]:
```suggestion
### Resource provider behavior
```

File: docs/content/reference/resource-schema/resource-policies/_index.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+#### Timeout times
```
Comment (AaronCrawfis) [inline docs/content/reference/resource-schema/resource-policies/_index.md]:
```suggestion
#### Timeout durations
```

File: docs/content/reference/resource-schema/resource-policies/_index.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+- Default asynchronous API timeout : 120 seconds
+
+#### Asynchronous operation retry policy
+
+Each resource type controller decides whether it will retry to process the operation when it fails, but the current async operation controller immediately marks any errors as a `Failed` operation except for the following scenario:
+
+- Panic happens while processing the async operation by code defect (retry once, then mark the operation Failed)
+- RP process exits by unexpected process crashes (such as node failure, memory leak) and redeploying Radius. (retry once , then mark the operation Failed)
+
+##### Applications.Core resource provider
```
Comment (AaronCrawfis) [inline docs/content/reference/resource-schema/resource-policies/_index.md]: We will need to make sure to have a process of how to update this table whenever a new resource is added to the app model. This page will quickly go out of date unless the dev docs say to update this when adding a new resource. Best solution would be to eventually auto-generate it. We should open an issue to track this.

**Review by AaronCrawfis — CHANGES_REQUESTED** [review]

File: docs/content/reference/resource-schema/resource-policies/_index.md (hunk 7 above)
Comment (jasonviviano) [reply docs/content/reference/resource-schema/resource-policies/_index.md]: Created this in the docs repo: https://github.com/project-radius/docs/issues/290

File: docs/content/reference/resource-schema/resource-policies/_index.md (hunk 1 above)
Comment (jasonviviano) [reply docs/content/reference/resource-schema/resource-policies/_index.md]: Created the `reference/api` page and moved the timeout policy page let me know if this makes more sense for you.

File: docs/content/reference/api/_index.md [hunk 8]
```diff
@@ -0,0 +1,9 @@
+---
+type: docs
+title: "Radius API reference"
+linkTitle: "rad API"
```
Comment (AaronCrawfis) [inline docs/content/reference/api/_index.md]:
```suggestion
linkTitle: "Radius API"
```

`rad` is only for the CLI

File: docs/content/reference/api/_index.md [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+description: "Detailed reference documentation on the Radius API"
+weight: 100
```
Comment (AaronCrawfis) [inline docs/content/reference/api/_index.md]:
```suggestion
weight: 400
```

File: docs/content/reference/api/resource-policies/_index.md [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+There are two main Radius services that will provide the resources described below:
+
+- [Deployment Engine]({{< ref "architecture.md#deployment-engine" >}})
+- [Resource Provider]({{< ref "architecture.md#radius-resource-provider" >}})
+
+### Deployment engine behavior
+
+| Operation | Resource type | Server Timeout (Seconds) | Async Operation retry condition|
+|------|:--------:|-------------|---------|------------|
+| Default retry timeout for resource deployment | The name of your resource. | 12.5 minutes per resource | Retries on 5xx errors, 50 retry requests
```
Comment (AaronCrawfis) [inline docs/content/reference/api/resource-policies/_index.md:20]: In this table you say "name of your resource", but the column header says "Resource type". I'd be specific on which one you're referencing

File: docs/content/reference/api/resource-policies/_index.md [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+---
+
+## Radius services
+
+There are two main Radius services that will provide the resources described below:
+
+- [Deployment Engine]({{< ref "architecture.md#deployment-engine" >}})
+- [Resource Provider]({{< ref "architecture.md#radius-resource-provider" >}})
+
+### Deployment engine behavior
```
Comment (AaronCrawfis) [inline docs/content/reference/api/resource-policies/_index.md]: I'd bump this header up from h3 to h2, as well as promote every header below this. I try not to have only one h2 on a given page. It's better to have multiple H2s and not nest too deep into h4/h5

**Review by AaronCrawfis — CHANGES_REQUESTED** [review]

**Review by AaronCrawfis — APPROVED** [review]

**Pull request merged**